tests/test_writers.py

```diff
@@ -316,3 +320,92 @@ def test_mysql_types(self):
         '''
         with self.mysql_engine.connect() as conn:
             self.SqlWriter_types_test(conn, is_mysql=True)
+
+
+def make_test_cases(cls):
+    """For each '_test_*' method on the class create a test case for each sql engine"""
+    engines = ('mysql_engine', 'postgres_engine')
+    methods = inspect.getmembers(cls, predicate=inspect.ismethod)
+    for name, method in methods:
+        if name.startswith('_test_'):
+            def _make_test_cases(test_method):
+                # closure to make sure test_method is the right method
+                for engine in engines:
+                    def test(self):
+                        with getattr(self, engine).connect() as conn:
```

Comment:
This will not work as expected. By the time the method gets called, the name `engine` will reference the last item in the `engines` tuple. In other words, `engine` will be `'postgres_engine'` for all tests generated by this `for` loop.

One way to work around this is to use a function to create the test method. Something like this:
```py
def make_test(engine):
    def test(self):
        with getattr(self, engine)...
            ...
    return test
```
Then in the loop you can do `test = make_test(engine)`. [Explanation](https://eev.ee/blog/2011/04/24/gotcha-python-scoping-closures/).